db/migrate/20221113212453_backfill_applicant_has_national_insurance_number.rb

```diff
@@ -0,0 +1,10 @@
+class BackfillApplicantHasNationalInsuranceNumber < ActiveRecord::Migration[7.0]
+  disable_ddl_transaction!
+
+  def up
+    Applicant.unscoped.where.not(national_insurance_number: nil).in_batches do |relation|
+      relation.update_all(has_national_insurance_number: true)
+      sleep(0.01) # throttle
```

Comment:
Got this idea from the [strong migrations gem](https://github.com/ankane/strong_migrations#backfilling-data). The idea i believe, from a quick google, is to give the db a chance to release any buffers and memory between batches.

😳 Just check and this could potentially cause a 324+ second deployment as there are 32414 records. I guess we should not throttle and instead do it out of hours to avoid potential impact on users?!

Reply:
Have you considered de-coupling the backfill of data from the change?

For instance, we could introduce a method like this temporarily

```ruby
def has_national_insurance_number
  super || !!national_insurance_number
end
```

then we can backfill the data in a separate change, then finally remove this method.

What do you think?